### PR TITLE
Unit tests say signal gets the correct list passed with it

### PR DESCRIPTION
**Pull request by MrMelbert**
## About The Pull Request

#63939 broke this once, #69844 almost broke this again, so I decided to unit test this

`COMSIG_MOB_SAY` signal handlers override the arglist it is passed to change parameters of the say. Such as changing the message (adding a stutter or capslocking), changing language, adding spans, etc. 

This means it needs to get the arglist, exactly, by reference - not a copy of it or anything, otherwise the overriding parameter aspect won't work and a lot of say stuff breaks 

In the meanwhile I learned some byond knowledge, in that every child call of a proc-call has its own arglist, which is its own reference. So, attempting to use a dummy mob instead of doing it on the `/living` level results in two different refs. 

## Why It's Good For The Game

Making sure message modifiers work is pretty neat

### CODE REVIEW

File: code/modules/mob/living/living_say.dm [hunk 1]
```diff
@@ -231,7 +231,14 @@ GLOBAL_LIST_INIT(message_modes_stat_limits, list(
...
 
+	#ifdef UNIT_TESTS
+	// Saves a ref() to our arglist specifically.
+	// We do this because we need to check that COMSIG_MOB_SAY is getting EXACTLY this list.
+	last_say_args_ref = REF(args)
+	#endif
+
 	// Leaving this here so that anything that handles speech this way will be able to have spans affecting it and all that.
+	// Make sure the arglist is passed exactly - don't pass a copy of it. Say signal handlers will modify some of the parameters.
 	var/sigreturn = SEND_SIGNAL(src, COMSIG_MOB_SAY, args, message_range)
```
Comment (timothymtorres) [inline code/modules/mob/living/living_say.dm:242]: Appending args by adding them to the end of the parameters does not work.  I tested this on tg master (not my hivemind branch)

Code_yXIlmidcjY

The `message_range` arg does not get added to args.  It gets thrown away.

File: code/modules/unit_tests/say.dm [hunk 2]
```diff
@@ -21,3 +21,25 @@
...
+
+	RegisterSignal(dummy, COMSIG_MOB_SAY, .proc/check_say)
+	dummy.say(test_message)
+
+/datum/unit_test/say_signal/proc/check_say(mob/living/source, list/say_args)
+	SIGNAL_HANDLER
+
+	TEST_ASSERT_EQUAL(say_args[SPEECH_MESSAGE], test_message, "Say signal's first arg of the arglist wasn't the expected message.")
+
+	TEST_ASSERT_EQUAL(REF(say_args), source.last_say_args_ref, "Say signal didn't get the argslist of say as a reference. \
```
Comment (Mothblocks) [inline code/modules/unit_tests/say.dm:40]: The juggling here shouldn't be necessary. if `REF(say_args) == source.last_say_args_ref`, then the lists would equal each other too.

**Review by Mothblocks — CHANGES_REQUESTED** [review]
This is my first time seeing people mutate args through signals and honestly I hate it, no wonder this keeps breaking, it's insane

**Comment by Mothblocks** [comment]
Mutating lists through signals is whatever but specifically doing it to args is insane

**Pull request merged**